gzip.go

```diff
@@ -405,7 +405,11 @@ func (z *Writer) Write(p []byte) (int, error) {
 		if len(z.currentBuffer) == z.blockSize {
 			z.compressCurrent(false)
 			if err := z.checkError(); err != nil {
-				return len(p) - len(q) - length, err
+				numBytes := len(p) - len(q) - length
+				if numBytes < 0 {
+					numBytes = 0
+				}
+				return numBytes, err
```

Comment:
This should never be negative and work fine as well.
```suggestion
				return len(p) - len(q), err
```